mesonbuild/environment.py

```diff
@@ -477,7 +480,7 @@ def create_new_coredata(self, options):
         self.first_invocation = True
 
     def is_cross_build(self):
-        return not self.machines.matches_build_machine(MachineChoice.HOST)
+        return self.have_cross_file
```

Comment:
I don't think we need anything more than:
```python
return self.coredata.cross_file is not None
```

Reply:
Yeah that is better.